### PR TITLE
[W6.4h][W11-B2]Lai Qi Wei

### PR DESCRIPTION
**Pull request by qwlai**
Ready for review.

Added Printable interface
Name, Phone, Email, Address now implements Printable
ReadOnlyPerson internal methods now calls getPrintableString(Printable... printables)
Added JUnit test for getPrintableString()

### CODE REVIEW

File: src/seedu/addressbook/data/person/Printable.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package seedu.addressbook.data.person;
+
+/**
+ * An interface that represents printable entities
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Printable.java:4]: :+1: for the appropriate comments

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 2]
```diff
@@ -36,50 +38,45 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
...
-        }
-        builder.append(getEmail())
-                .append(" Address: ");
-        if (getAddress().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getAddress())
-                .append(" Tags: ");
+        final StringJoiner joiner = new StringJoiner(" ");
+        joiner.add(getPrintableString(getName(), getPhone(), getEmail(), getAddress()));
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:42]: Good use of string joiner!

File: test/java/seedu/addressbook/data/PrintableTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                new Email("test@gmail.com", false),
+                new Address("321 Street", false),
+                new UniqueTagList(new Tag("tea"), new Tag("happy")));
+    }
+
+    /**
+     * Asserts that the details of the person is the same.
+     */
+    @Test
+    public void getPrintableString_compareWithExpectedString_printedCorrectly() {
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/PrintableTest.java:35]: :+1: for updating the tests as well.

**Review by okkhoy — COMMENTED** [review]
Good job on this one!

**Comment by okkhoy** [comment]
@qwlai 
Some comments added. Please ack & close the PR after reading comments.

**Comment by qwlai** [comment]
Thanks for reviewing.

**Pull request closed without merging**